migration-tool/src/main/java/software/amazon/awssdk/migration/internal/recipe/HttpSettingsToHttpClient.java

```diff
@@ -15,57 +15,444 @@
 
 package software.amazon.awssdk.migration.internal.recipe;
 
+import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
 import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
 import java.util.Optional;
 import java.util.Set;
+import org.jetbrains.annotations.NotNull;
+import org.openrewrite.Cursor;
 import org.openrewrite.ExecutionContext;
 import org.openrewrite.Recipe;
+import org.openrewrite.Tree;
 import org.openrewrite.TreeVisitor;
 import org.openrewrite.java.JavaIsoVisitor;
 import org.openrewrite.java.tree.Expression;
 import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.JContainer;
+import org.openrewrite.java.tree.JRightPadded;
+import org.openrewrite.java.tree.JavaType;
 import org.openrewrite.java.tree.Space;
+import org.openrewrite.java.tree.TypeUtils;
+import org.openrewrite.marker.Markers;
 import software.amazon.awssdk.annotations.SdkInternalApi;
 import software.amazon.awssdk.core.client.config.ClientOverrideConfiguration;
+import software.amazon.awssdk.http.apache.ApacheHttpClient;
+import software.amazon.awssdk.http.nio.netty.NettyNioAsyncHttpClient;
+import software.amazon.awssdk.migration.internal.utils.SdkTypeUtils;
+import software.amazon.awssdk.utils.CollectionUtils;
+import software.amazon.awssdk.utils.Logger;
+import software.amazon.awssdk.utils.Pair;
 
+/**
+ * This recipe moves the HTTP settings such as maxConnections configured on v1 ClientConfiguration to v2 ApacheHttpClient builder
+ * for sync SDK client or to v2 NettyNioAsyncHttpClient builder for async SDK client.
+ * <p>
+ * {@snippet :
+ *         ClientConfiguration clientConfiguration = new ClientConfiguration()
```

Comment:
Will the movement of configs from `ClientConfiguration` work when the scopes are not the same? 

```
public class MyClass {
    private final ClientConfiguration config = ...;

    public void run() {
        AmazonSQS sqs = AmazonSQSClient.builder().withConfiguration(config)...
```

Reply:
It should handle it, let me write a test for it

Reply:
Added more tests and verified it worked,